src/TelemetryChannels/ServerTelemetryChannel/Shared/Implementation/TransmissionStorage.cs

```diff
@@ -217,7 +242,7 @@ private IPlatformFile CreateTemporaryFile()
         private IEnumerable<IPlatformFile> GetTransmissionFiles()
         {
             IEnumerable<IPlatformFile> newFiles = this.folder.GetFiles();
-            newFiles = newFiles.Where(f => f.Extension == TransmissionFileExtension);
+            newFiles = newFiles.Where(f => f.Extension == TransmissionFileExtension && !this.badFiles.ContainsKey(f.Name));
```

Comment:
Also `EnsureSizeIsCalculated` will not include bad files now? So the folder can grow infinitely... Maybe those files can be included in size calculation, but not for the transmission.